File: solutions/vbnet/triangle/1/Triangle.vb
Imports System

Public Module Triangle
    Public Function IsScalene(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso Not IsEquilateral(side1, side2, side3) AndAlso Not IsIsosceles(side1, side2, side3)
    End Function

    Public Function IsIsosceles(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso (side1 = side2 OrElse side2 = side3 OrElse side3 = side1)
    End Function

    Public Function IsEquilateral(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso Math.Pow(side1, 2) + Math.Pow(side2, 2) = 2 * Math.Pow(side3, 2)
    End Function

    Private Function IsTriangle(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return (side1 > 0 AndAlso side2 > 0 AndAlso side3 > 0) AndAlso side1 + side2 >= side3 AndAlso side2 + side3 >= side1 AndAlso side3 + side1 >= side2
    End Function
End Module

File: solutions/euphoria/triangle/1/triangle.ex
include std/sequence.e
include std/sort.e
include std/math.e

function unique_sides(sequence sides)
  sides = sort(sides)
  return length(remove_dups(sides)) + 9*(sides[1] + sides[2] < sides[3] or sum(sides <= 0))
end function

public function is_equilateral(sequence sides)
  return unique_sides(sides) = 1
end function

public function is_isosceles(sequence sides)
  return unique_sides(sides) <= 2
end function

public function is_scalene(sequence sides)
  return unique_sides(sides) = 3
end function

  